perf(Auto Attendance): Use background job to process auto attendance

asmitahase opened the pull request:
The auto-attendance job is pretty fast as is. Takes less than 30 seconds to process a few day's attendance for a thousand employees, in an ideal situation. Although if there's any custom code that then does some other operations while submitting attendance, the time can add up, specially if something goes wrong and the whole month's attendance is attempted to process at the end of the month to process payroll.
For these cases, it's best to hand that job over to a background worker instead of trying to process it in realtime and being greeted with a timeout error. Hence this attempt.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Manual trigger for shift attendance processing from Shift Type settings.
  * Large attendance batches (1000+ records) now enqueue for background processing with job/status links.

* **Behavior Changes**
  * Notifications now display dynamic, response-driven messages.
  * Processing is skipped when shift configuration is invalid and shows clearer status/error handling.

* **Tests**
  * Added test to verify background job creation for large attendance batches.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ruchamahabal reviewed (commented):
The hourly job that runs is already a BG job that processes it for every shift? If you want to make this change just for the button in shift type, create a separate whitelisted method for it and enqueue job in it instead?

asmitahase commented:
Makes sense

asmitahase committed:
perf(Auto Attendance): Use background job to process auto attendance for large number of checkin logs

ruchamahabal commented on hrms/hr/doctype/shift_type/shift_type.py:
```diff
@@ -98,15 +98,34 @@ def unlinked_checkins_exist(self):
-		):
+		)
+
+	def process_auto_attendance(self):
```
BTW, removing whitelisting for it will break APIs for people with integrations/client scripts calling this method. You can accept a param in the function to check if it was manually called maybe and then enqueue if reqd.

asmitahase committed:
fix: preserve whitelisted method
fix: prevent duplicate job queueing
fix: show job link in message to moniter status

asmitahase committed:
fix: log any exceptions encountered during the request
chore: flag to direct flow during tests
chore: typo

asmitahase committed:
fix: return error feedback

asmitahase committed:
test: background job creation for large checkins
